### PR TITLE
Fix start minimized bug

### LINKED ISSUE #2061: Mac App minimize at startup
I package artifacts on Mac and create ControlCatalog Mac app [Dropbox Link here](https://www.dropbox.com/s/vq565nh8eu0pr2s/ControlCatalog.zip?dl=0)

- It minimizes immediately at startup. 
- closing the window does not quit the app.

### PR DESCRIPTION
**Pull request by jeffreye**
This template is not intended to be prescriptive, but to help us review pull requests it would be useful if you included as much of the following information as possible:

- What does the pull request do?
Fix #2061 
- What is the current behavior?
1. Startup and set _lastWindowState, which is minimized after packaging
2. By clicking close button, it closes sometimes and other times not
- What is the updated/expected behavior with this PR?
1. startup with Normal state
2. implement Minimize and Maximize state changed event
3. close the main window and close the application
- How was the solution implemented (if it's not obvious)?
1. startup with Normal state
2. keep cancellation token alive util native application loop exit

### CODE REVIEW

File: src/Avalonia.Native/PlatformThreadingInterface.cs [hunk 1]
```diff
@@ -68,8 +68,6 @@ public void RunLoop(CancellationToken cancellationToken)
                     lock (l)
                     {
                         cancellation?.Cancel();
-                        cancellation?.Dispose();
```
Comment (kekekeks) [inline src/Avalonia.Native/PlatformThreadingInterface.cs]: Why these lines are removed? This causes a memory leak.

**Comment by jeffreye** [comment]
The cancellation token need to be kept alive util it notify native loop. And it would eventually be disposed in the finally block.

**Review by kekekeks — APPROVED** [review]

**Comment by danwalmsley** [comment]
@jeffreye @kekekeks I think the correct fix for the minimise should have been to initialise the value to normal here:

https://github.com/jeffreye/Avalonia/blob/16bc1097da0c6214385d4a728105697168afac09/native/Avalonia.Native/src/OSX/window.mm#L402

**Comment by kekekeks** [comment]
There were other valuable changes there, so I've merged it anyway. Feel free to change it from Minimized to Normal there.

**Comment by jeffreye** [comment]
Yes, and I found the lastWindowState is only being
 used in the OnResized method. But OnResized never get called.

**Pull request merged**